From 75116bb1afa9572008468d8b4bf9eb9fda76e014 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Thu, 20 Dec 2018 11:22:34 -0700
Subject: [PATCH 2/3] dma: imx-dma: Cast sg_dma_len() to size_t

Avoids Clang warning:

drivers/dma/imx-dma.c:287:8: warning: comparison of distinct pointer types ('typeof (d->len) *' (aka 'unsigned long *') and 'typeof (((sg)->dma_length)) *' (aka 'unsigned int *')) [-Wcompare-distinct-pointer-types]
        now = min(d->len, sg_dma_len(sg));
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~
./include/linux/kernel.h:878:19: note: expanded from macro 'min'
 #define min(x, y)       __careful_cmp(x, y, <)
                         ^~~~~~~~~~~~~~~~~~~~~~
./include/linux/kernel.h:869:24: note: expanded from macro '__careful_cmp'
        __builtin_choose_expr(__safe_cmp(x, y), \
                              ^~~~~~~~~~~~~~~~
./include/linux/kernel.h:859:4: note: expanded from macro '__safe_cmp'
                (__typecheck(x, y) && __no_side_effects(x, y))
                 ^~~~~~~~~~~~~~~~~
./include/linux/kernel.h:845:29: note: expanded from macro '__typecheck'
                (!!(sizeof((typeof(x) *)1 == (typeof(y) *)1)))
                           ~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~
1 warning generated.

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 drivers/dma/imx-dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/imx-dma.c b/drivers/dma/imx-dma.c
index c2fff3f6c9ca..2382237736c2 100644
--- a/drivers/dma/imx-dma.c
+++ b/drivers/dma/imx-dma.c
@@ -285,7 +285,7 @@ static inline int imxdma_sg_next(struct imxdma_desc *d)
 	struct scatterlist *sg = d->sg;
 	unsigned long now;
 
-	now = min(d->len, sg_dma_len(sg));
+	now = min(d->len, (size_t)sg_dma_len(sg));
 	if (d->len != IMX_DMA_LENGTH_LOOP)
 		d->len -= now;
 
-- 
2.20.1

